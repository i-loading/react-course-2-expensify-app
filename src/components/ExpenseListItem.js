import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h4>
        <i>{description}</i>
      </h4>
    </Link>
    <p>Amount: {amount}</p>
    <p>Created at: {createdAt}</p>
  </div>
);

export default ExpenseListItem;
