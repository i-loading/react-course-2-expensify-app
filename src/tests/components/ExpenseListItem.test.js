import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "./../../components/ExpenseListItem";
import expense from "../fixtures/expense";

test("Should render correct ExpenseListItem component with fixture data", () => {
  const wrapper = shallow(<ExpenseListItem {...expense[0]} />);
  expect(wrapper).toMatchSnapshot();
});
