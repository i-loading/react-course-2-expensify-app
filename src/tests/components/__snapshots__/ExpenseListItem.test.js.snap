// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correct ExpenseListItem component with fixture data 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h4>
      <i>
        Gum
      </i>
    </h4>
  </Link>
  <p>
    Amount: 
    $1.95
  </p>
  <p>
    Created at: 
    January 1st, 1970
  </p>
</div>
`;
